OWML.ModHelper.Menus/ModPopupInput.cs

```diff
@@ -1,15 +1,28 @@
 ﻿using OWML.Common.Menus;
 using OWML.ModHelper.Events;
+using OWML.ModHelper.Input;
+using UnityEngine;
 using UnityEngine.UI;
 
 namespace OWML.ModHelper.Menus
 {
     public abstract class ModPopupInput<T> : ModInput<T>
     {
         protected readonly TwoButtonToggleElement ToggleElement;
+        protected ModCommandListener CommandListener;
 
         public override bool IsSelected => ToggleElement.GetValue<bool>("_amISelected");
 
+        protected virtual void SetupCommands()
```

Comment:
Calling this multiple tines will cause unwanted side effects, is that OK? Maybe make it private and call from Initialize?

Reply:
Ah, you're right. Gotta add a null check

Reply:
Actually, yeah, the way it's done there is no point in making it protected virtual

Reply:
Also move this method below the constructor

Reply:
Idea behind making it protected was to allow override, I don't think you can override private methods, can you?

Reply:
Actually, tha made me think that I should add command check to OnOpenCommand, since CommandListener is protected

Reply:
and yeah, I probably can make setup private since CommandListener is protected either way

Reply:
same is true for ModMenuWithSelectables

Reply:
👍 methods should be below constructors, tho

Reply:
whoops, somehow forgot that